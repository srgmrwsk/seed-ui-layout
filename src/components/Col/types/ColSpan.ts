export type ColSpan = boolean | number | 'auto';
