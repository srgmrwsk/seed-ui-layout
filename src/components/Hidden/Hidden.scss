.seed-hidden {
  display: contents;

  &_xs-down {
    @include bp-down(xs, $grid-breakpoints) {
      display: none;
    }
  }

  &_sm {
    @include bp-only(sm, $grid-breakpoints) {
      display: none;
    }
  }

  &_sm-down {
    @include bp-down(sm, $grid-breakpoints) {
      display: none;
    }
  }

  &_sm-up {
    @include bp-up(sm, $grid-breakpoints) {
      display: none;
    }
  }

  &_md {
    @include bp-only(md, $grid-breakpoints) {
      display: none;
    }
  }

  &_md-down {
    @include bp-down(md, $grid-breakpoints) {
      display: none;
    }
  }

  &_md-up {
    @include bp-up(md, $grid-breakpoints) {
      display: none;
    }
  }

  &_lg {
    @include bp-only(lg, $grid-breakpoints) {
      display: none;
    }
  }

  &_lg-down {
    @include bp-down(lg, $grid-breakpoints) {
      display: none;
    }
  }

  &_lg-up {
    @include bp-up(lg, $grid-breakpoints) {
      display: none;
    }
  }

  &_xl {
    @include bp-only(xl, $grid-breakpoints) {
      display: none;
    }
  }

  &_xl-down {
    @include bp-down(xl, $grid-breakpoints) {
      display: none;
    }
  }

  &_xl-up {
    @include bp-up(xl, $grid-breakpoints) {
      display: none;
    }
  }

  &_xxl-up {
    @include bp-up(xxl, $grid-breakpoints) {
      display: none;
    }
  }
}
