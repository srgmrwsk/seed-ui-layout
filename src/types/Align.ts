export type Align = 'start' | 'end' | 'center' | 'baseline' | 'stretch';
