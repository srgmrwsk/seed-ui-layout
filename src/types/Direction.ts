export type Direction = 'column' | 'column-reverse' | 'row' | 'rew-reverse';
