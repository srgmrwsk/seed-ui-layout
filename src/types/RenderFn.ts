import React from 'react';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export type RenderFn = (component: React.ReactElement, ...args: []) => any;
