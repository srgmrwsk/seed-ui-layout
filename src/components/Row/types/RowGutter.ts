import { Sizing } from '../../../types/Sizing';

export type RowGutter = Sizing | [Sizing, Sizing];
