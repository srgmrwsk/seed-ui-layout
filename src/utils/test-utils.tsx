import React from 'react';
import { render } from '@testing-library/react';
import { renderToStaticMarkup } from 'react-dom/server';
import { configureAxe } from 'jest-axe';

import { RenderFn } from '../types/RenderFn';

export const axe = configureAxe({
  rules: {
    // disable landmark rules when testing isolated components
    region: { enabled: false },
  },
});

export function create(
  ...args: Parameters<RenderFn>
): ChildNode | HTMLCollection | null {
  const { container } = render(...args);
  return container.children.length > 1
    ? container.children
    : container.firstChild;
}
export function renderToHtml(component: React.ReactElement): string {
  return renderToStaticMarkup(component);
}
