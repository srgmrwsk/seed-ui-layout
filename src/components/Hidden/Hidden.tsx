import React from 'react';
import cx from 'classnames';

import { getBooleanStyles } from '../../utils/get-boolean-styles';

export interface HiddenProps
  extends Omit<React.AllHTMLAttributes<HTMLElement>, 'as'> {
  as?: React.ElementType;
  xsDown?: boolean;
  sm?: boolean;
  smDown?: boolean;
  smUp?: boolean;
  md?: boolean;
  mdDown?: boolean;
  mdUp?: boolean;
  lg?: boolean;
  lgDown?: boolean;
  lgUp?: boolean;
  xl?: boolean;
  xlUp?: boolean;
  xlDown?: boolean;
  xxlUp?: boolean;
}

function Hidden(
  {
    as: As = 'div',
    xsDown = false,
    sm = false,
    smDown = false,
    smUp = false,
    md = false,
    mdDown = false,
    mdUp = false,
    lg = false,
    lgDown = false,
    lgUp = false,
    xl = false,
    xlDown = false,
    xlUp = false,
    xxlUp = false,
    className,
    children,
  }: HiddenProps,
  ref: React.Ref<HTMLElement>,
): JSX.Element {
  return (
    <As
      ref={ref}
      className={cx(
        'seed-hidden',
        getBooleanStyles('seed-hidden_xs-down', xsDown),
        getBooleanStyles('seed-hidden_sm', sm),
        getBooleanStyles('seed-hidden_sm-down', smDown),
        getBooleanStyles('seed-hidden_sm-up', smUp),
        getBooleanStyles('seed-hidden_md', md),
        getBooleanStyles('seed-hidden_md-down', mdDown),
        getBooleanStyles('seed-hidden_md-up', mdUp),
        getBooleanStyles('seed-hidden_lg', lg),
        getBooleanStyles('seed-hidden_lg-down', lgDown),
        getBooleanStyles('seed-hidden_lg-up', lgUp),
        getBooleanStyles('seed-hidden_xl', xl),
        getBooleanStyles('seed-hidden_xl-down', xlDown),
        getBooleanStyles('seed-hidden_xl-up', xlUp),
        getBooleanStyles('seed-hidden_xxl-up', xxlUp),
        className,
      )}
    >
      {children}
    </As>
  );
}

Hidden.displayName = 'Hidden';

export default React.forwardRef(Hidden);
