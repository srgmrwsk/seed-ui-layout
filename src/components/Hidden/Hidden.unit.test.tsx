import React from 'react';
import { renderToStaticMarkup } from 'react-dom/server';

import { RenderFn } from '../../types/RenderFn';
import { axe, create } from '../../utils/test-utils';

import Hidden, { HiddenProps } from './Hidden';

describe('Hidden', () => {
  const setup = (renderFn: RenderFn, props?: HiddenProps) =>
    renderFn(<Hidden {...props} />);

  it('should render correctly with default props', () => {
    const actual = setup(create);
    expect(actual).toMatchSnapshot();
  });

  it('should render correctly with as prop', () => {
    const actual = setup(create, { as: 'section' });
    expect(actual).toMatchSnapshot();
  });

  it('should render correctly with children prop', () => {
    const actual = setup(create, { children: <p>Element with text.</p> });
    expect(actual).toMatchSnapshot();
  });

  it('should render correctly with xsDown prop', () => {
    const actual = setup(create, { xsDown: true });
    expect(actual).toMatchSnapshot();
  });

  it('should render correctly with sm prop', () => {
    const actual = setup(create, { sm: true });
    expect(actual).toMatchSnapshot();
  });

  it('should render correctly with smDown prop', () => {
    const actual = setup(create, { smDown: true });
    expect(actual).toMatchSnapshot();
  });

  it('should render correctly with smUp prop', () => {
    const actual = setup(create, { smUp: true });
    expect(actual).toMatchSnapshot();
  });

  it('should render correctly with md prop', () => {
    const actual = setup(create, { md: true });
    expect(actual).toMatchSnapshot();
  });

  it('should render correctly with mdDown prop', () => {
    const actual = setup(create, { mdDown: true });
    expect(actual).toMatchSnapshot();
  });

  it('should render correctly with mdUp prop', () => {
    const actual = setup(create, { mdUp: true });
    expect(actual).toMatchSnapshot();
  });

  it('should render correctly with lg prop', () => {
    const actual = setup(create, { lg: true });
    expect(actual).toMatchSnapshot();
  });

  it('should render correctly with lgDown prop', () => {
    const actual = setup(create, { lgDown: true });
    expect(actual).toMatchSnapshot();
  });

  it('should render correctly with lgUp prop', () => {
    const actual = setup(create, { lgUp: true });
    expect(actual).toMatchSnapshot();
  });

  it('should render correctly with xl prop', () => {
    const actual = setup(create, { xl: true });
    expect(actual).toMatchSnapshot();
  });

  it('should render correctly with xlDown prop', () => {
    const actual = setup(create, { xlDown: true });
    expect(actual).toMatchSnapshot();
  });

  it('should render correctly with xlUp prop', () => {
    const actual = setup(create, { xlUp: true });
    expect(actual).toMatchSnapshot();
  });

  it('should render correctly with xxlUp prop', () => {
    const actual = setup(create, { xxlUp: true });
    expect(actual).toMatchSnapshot();
  });

  it('should meet accessibility guidelines', async () => {
    const wrapper = setup(renderToStaticMarkup);
    const actual = await axe(wrapper);
    expect(actual).toHaveNoViolations();
  });
});
