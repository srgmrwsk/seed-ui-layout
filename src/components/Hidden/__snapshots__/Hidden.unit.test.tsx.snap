// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hidden should render correctly with as prop 1`] = `
<section
  class="seed-hidden"
/>
`;

exports[`Hidden should render correctly with children prop 1`] = `
<div
  class="seed-hidden"
>
  <p>
    Element with text.
  </p>
</div>
`;

exports[`Hidden should render correctly with default props 1`] = `
<div
  class="seed-hidden"
/>
`;

exports[`Hidden should render correctly with lg prop 1`] = `
<div
  class="seed-hidden seed-hidden_lg"
/>
`;

exports[`Hidden should render correctly with lgDown prop 1`] = `
<div
  class="seed-hidden seed-hidden_lg-down"
/>
`;

exports[`Hidden should render correctly with lgUp prop 1`] = `
<div
  class="seed-hidden seed-hidden_lg-up"
/>
`;

exports[`Hidden should render correctly with md prop 1`] = `
<div
  class="seed-hidden seed-hidden_md"
/>
`;

exports[`Hidden should render correctly with mdDown prop 1`] = `
<div
  class="seed-hidden seed-hidden_md-down"
/>
`;

exports[`Hidden should render correctly with mdUp prop 1`] = `
<div
  class="seed-hidden seed-hidden_md-up"
/>
`;

exports[`Hidden should render correctly with sm prop 1`] = `
<div
  class="seed-hidden seed-hidden_sm"
/>
`;

exports[`Hidden should render correctly with smDown prop 1`] = `
<div
  class="seed-hidden seed-hidden_sm-down"
/>
`;

exports[`Hidden should render correctly with smUp prop 1`] = `
<div
  class="seed-hidden seed-hidden_sm-up"
/>
`;

exports[`Hidden should render correctly with xl prop 1`] = `
<div
  class="seed-hidden seed-hidden_xl"
/>
`;

exports[`Hidden should render correctly with xlDown prop 1`] = `
<div
  class="seed-hidden seed-hidden_xl-down"
/>
`;

exports[`Hidden should render correctly with xlUp prop 1`] = `
<div
  class="seed-hidden seed-hidden_xl-up"
/>
`;

exports[`Hidden should render correctly with xsDown prop 1`] = `
<div
  class="seed-hidden seed-hidden_xs-down"
/>
`;

exports[`Hidden should render correctly with xxlUp prop 1`] = `
<div
  class="seed-hidden seed-hidden_xxl-up"
/>
`;
